src/lib.rs

```diff
@@ -56,11 +56,36 @@ impl Cli {
 
         let delete_file = run::generate_delete_file(tmp_dir.path())?;
 
-        run::start_shell(config, tmp_dir.path())?;
+        let subprocesses = run::start_subprocesses(config, tmp_dir.path());
 
-        run::clean_up(delete_file, tmp_dir, self.worktree_branch.clone())?;
+        let shell = run::start_shell(config, tmp_dir.path());
 
-        Ok(())
+        #[cfg(windows)]
+        if config.editor.is_some() {
+            unsafe {
+                windows_sys::Win32::System::Console::FreeConsole();
+            }
+        }
+
+        let res = match shell {
+            Ok(mut child) => child.wait(),
+            Err(err) => {
+                bail!("{}", err)
```

Comment:
Since you bailed here. The cleanup won't happen. `bail!()` is the equivalent of doing `return Err(...);`